### ISSUE TITLE
End of long line is occluded by scrollbar

### ISSUE THREAD

**Issue by iLynxcat** [zed-industries/zed#14966]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

With a line that is at least several characters wider than the editor's viewport, jumping to the end (via ⌘→ on Mac) will move the cursor to the end as expected, but if the scrollbar is visible Zed does not take it into account and the cursor and last character are occluded by the scrollbar. The mouse must be used to pan the viewport right to display the end of the line.

### Environment

Zed: v0.144.4 (Zed)
OS: macOS 15.0.0
Memory: 8 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

https://github.com/user-attachments/assets/477c5f01-7559-4ad2-b06f-b49db7bf3241

**Comment by gavinmorrow** [comment]
I think this is a duplicate of https://github.com/zed-industries/zed/issues/11915.

**Comment by SomeoneToIgnore** [comment]
Indeed, duplicate of https://github.com/zed-industries/zed/issues/11915
